### PR TITLE
Add crypto built-in function implementations

### PR DESCRIPTION
**Pull request by anderseknert**
* cljs: hmac + hashing functions
* cls: crypto.x509.parse_rsa_private_key

Signed-off-by: Anders Eknert <anders@eknert.com>

### CODE REVIEW

File: src/main/cljc/jarl/builtins/crypto.cljc [hunk 1]
```diff
@@ -0,0 +1,95 @@
+(ns jarl.builtins.crypto
+  (:require [jarl.encoding.hex :as hex]
+            [jarl.builtins.utils :refer [str->bytes]]
+            #?(:clj  [jarl.encoding.base64 :as base64])
+            #?(:clj  [clojure.string :as str])
+            #?(:cljs [goog.crypt :as crypt]))
```
Comment (johanfylling) [inline src/main/cljc/jarl/builtins/crypto.cljc:6]: Is this a transitive dependency from somewhere?

**Review by johanfylling — APPROVED** [review]
👍

File: README.md [hunk 2]
```diff
@@ -262,17 +262,19 @@ TODO: Use https://github.com/henryw374/cljc.java-time or https://github.com/juxt
...
+| `crypto.md5`                                |  ✅  |  ✅   |    ✅    |
+| `crypto.sha1`                               |  ✅  |  ✅   |    ✅    |
+| `crypto.sha256`                             |  ✅  |  ✅   |    ✅    |
 | `crypto.x509.parse_and_verify_certificates` |  ❌  |  ❌   |    ❌    |
 | `crypto.x509.parse_certificate_request`     |  ❌  |  ❌   |    ❌    |
 | `crypto.x509.parse_certificates`            |  ❌  |  ❌   |    ❌    |
-| `crypto.x509.parse_rsa_private_key`         |  ❌  |  ❌   |    ❌    |
+| `crypto.x509.parse_rsa_private_key`         |  ✅  |  ❌   |    ❌    |
+
+NOTE: Contrary to the Clojure versions, the ClojureScript HMAC functions won't accept unicode (i.e. non-ascii) input
```
[the remark this answers is not in the document]
Comment (anderseknert) [reply README.md:277]: Indeed — see https://github.com/johanfylling/jarl/pull/101/files#diff-d9f6c09ca230b46030576fa96e15ba0673c37a6f03e70270c36857414b1d2e68R12

[the remark this answers is not in the document]
Comment (anderseknert) [reply README.md:277]: OK, that link seems useless. but see the diff in generator.clj

File: src/main/cljc/jarl/builtins/crypto.cljc (hunk 1 above)
Comment (anderseknert) [reply src/main/cljc/jarl/builtins/crypto.cljc:6]: ClojureScript is compiled using the Google Closure (yeah, quite a clash of names!) compiler, which provides the Closure library for "free". Those libraries are hence guaranteed to be present in any cljs project: https://www.clojurescript.org/about/closure

**Pull request merged**